### PR TITLE
Libvirt machine api

### PR DESCRIPTION
**Pull request by enxebre**
Move libvirt machines from k8s.io into openshift.io

We are transitioning the cluster API types from k8s.io over a machine.openshift.io group.
https://github.com/openshift/cluster-api/tree/91fca585a85b163ddfd119fd09c128c9feadddca
Although this give us a necessary level of autonomy today, we aim to consolidate back with upstream once there's a release of the API stable enough
For reference:
Backward compatible changes on machine api operator
[[openshift/machine-api-operator#185](https://github.com/openshift/machine-api-operator/pull/185)]([openshift/machine-api-operator#185](https://github.com/openshift/machine-api-operator/pull/185))
[[openshift/machine-api-operator#188](https://github.com/openshift/machine-api-operator/pull/188)]([openshift/machine-api-operator#188](https://github.com/openshift/machine-api-operator/pull/188))
[[openshift/machine-api-operator#191](https://github.com/openshift/machine-api-operator/pull/191)]([openshift/machine-api-operator#191](https://github.com/openshift/machine-api-operator/pull/191))
Libvirt actuator support for new group
[openshift/cluster-api-provider-libvirt#111](https://github.com/openshift/cluster-api-provider-libvirt/pull/111)

### CODE REVIEW

**Comment by enxebre** [comment]
cc @bison

**Comment by ingvagabund** [comment]
The only places where `sigs.k8s.io/cluster-api/` is imported is in deprecated code or in cluster object definitions.

File: pkg/asset/machines/master.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 		mpool.Set(ic.Platform.Libvirt.DefaultMachinePlatform)
 		mpool.Set(pool.Platform.Libvirt)
 		pool.Platform.Libvirt = &mpool
-		m.MachinesDeprecated, err = libvirt.Machines(clusterID.ClusterID, ic, &pool, "master", "master-user-data")
+		m.Machines, err = libvirt.Machines(clusterID.ClusterID, ic, &pool, "master", "master-user-data")
```
Comment (wking) [inline pkg/asset/machines/master.go]: This needs an associated change [here][1].

[1]: https://github.com/openshift/installer/blob/193a3f2cb7b8da375da90126ac4757254cdb17fe/pkg/asset/cluster/tfvars.go#L120

**Comment by wking** [comment]
[e2e-aws hung][1].

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1204/pull-ci-openshift-installer-master-e2e-aws/3509/build-log.txt

**Comment by wking** [comment]
[e2e-aws][1]:

```
Failing tests:

[Conformance][Area:Networking][Feature:Router] The default ClusterIngress should support default wildcard reencrypt routes through external DNS [Suite:openshift/conformance/parallel/minimal]
```

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1204/pull-ci-openshift-installer-master-e2e-aws/3520/build-log.txt

**Comment by wking** [comment]
[e2e-aws][1]:

```
Flaky tests:

[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should support existing directories when readOnly specified in the volumeSource [Suite:openshift/conformance/parallel] [Suite:k8s]

Failing tests:

[Conformance][Area:Networking][Feature:Router] The default ClusterIngress should support default wildcard reencrypt routes through external DNS [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance] oc new-app  should fail with a --name longer than 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance] oc new-app  should succeed with a --name of 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in docker build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in sti build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in docker build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in s2i build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  [Conformance] buildconfigs should have a default history limit set when created via the group api [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune builds after a buildConfig change [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune canceled builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune completed builds based on the successfulBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune errored builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune failed builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
```

/restest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1204/pull-ci-openshift-installer-master-e2e-aws/3530/build-log.txt

**Pull request merged**